02-express-tutorial/controllers/people.js

```diff
@@ -1,62 +1,61 @@
 let people = [];
 
 const getPeople = (req, res) => {
-    res.status(200).json({ success: true, data: people })
-  }
-  
-  const createPerson = (req, res) => {
-    const { name } = req.body
+    res.status(200).json({ success: true, data: people });
+};
+
+const createPerson = (req, res) => {
+    const { name } = req.body;
     if (!name) {
-      return res
-        .status(400)
-        .json({ success: false, msg: 'please provide name value' })
+        return res
+            .status(400)
+            .json({ success: false, msg: "please provide name value" });
     }
-    res.status(201).send({ success: true, person: name })
-  }
+    people.push({ id: people.length + 1, name });
+    res.status(201).send({ success: true, person: name });
+};
 
+const updatePerson = (req, res) => {
+    const { id } = req.params;
+    const { name } = req.body;
 
+    const person = people.find((person) => person.id === Number(id));
 
-  const updatePerson = (req, res) => {
-    const { id } = req.params
-    const { name } = req.body
-  
-    const person = people.find((person) => person.id === Number(id))
-  
     if (!person) {
-      return res
-        .status(404)
-        .json({ success: false, msg: `no person with id ${id}` })
-        people.push({ id: people.length + 1, name });
-  res.status(201).json({ success: true, person: name });
+        return res
+            .status(404)
+            .json({ success: false, msg: `no person with id ${id}` })
     }
-    
+
     const newPeople = people.map((person) => {
-      if (person.id === Number(id)) {
-        person.name = name
-      }
-      return person
-    })
-    res.status(200).json({ success: true, data: newPeople })
-  }
-  
-  const deletePerson = (req, res) => {
-    const person = people.find((person) => person.id === Number(req.params.id))
+        if (person.id === Number(id)) {
+            person.name = name;
+        }
+        return person;
+    });
+    res.status(200).json({ success: true, data: newPeople });
+};
+
+const deletePerson = (req, res) => {
+    const person = people.find((person) => person.id === Number(req.params.id));
     if (!person) {
-      return res
-        .status(404)
-        .json({ success: false, msg: `no person with id ${req.params.id}` })
+        return res
+            .status(404)
+            .json({
+                success: false,
+                msg: `no person with id ${req.params.id}`,
+            });
     }
     const newPeople = people.filter(
-      (person) => person.id !== Number(req.params.id)
-    )
-    people = newPeople;
-
-    return res.status(200).json({ success: true, data: newPeople })
-  }
-  
-  module.exports = {
+        (person) => person.id !== Number(req.params.id)
+    );
+
```

Comment:
With the latest commit, the following line was removed so the people array still has the deleted person.
people = newPeople;

Adding back the line should resolve the issue